client/app/vps/vps-task.service.js

```diff
@@ -0,0 +1,126 @@
+class VpsTaskService {
+    constructor ($http, $q, $rootScope, $timeout, $translate, CloudMessage) {
+        this.$http = $http;
+        this.$q = $q;
+        this.$rootScope = $rootScope;
+
+        this.$timeout = $timeout;
+        this.$translate = $translate;
+        this.CloudMessage = CloudMessage;
+
+        this.firstCall = true;
+        this.COMPLETED_TASK_PROGRESS = 100;
+    }
+
+    /*
+     * imported from vps services to avoid calling this services
+     * + _getTaskInPropgress
+     */
+
+    _getTaskInProgress (serviceName, type) {
+        return this.$http.get(["/sws/vps", serviceName, "tasks/uncompleted"].join("/"), {
+            serviceType: "aapi",
+            params: {
+                type: type
+            }
+        })
+        .then(data => data.data)
+        .catch(error => this.$q.reject(error.data));
+    }
+
+    /*
+     * subscribe : reset values and get pending tasks
+     *
+     */
+    subscribe (serviceName) {
+        this.firstCall = true;
+        this.getTasks(serviceName);
+    }
+
+    /*
+     * getTasks : retrieve task in progress
+     *
+     */
+    getTasks (serviceName) {
+        this._getTaskInProgress(serviceName)
+            .then(tasks => this.handleTasks(serviceName, tasks))
+            .catch(err => this.CloudMessage(err));
+    }
+
+    /*
+     * handleTasks : manage message to display and update value while there is task in progress
+     *
+     */
+    handleTasks (serviceName, tasks) {
+        this.flushMessages();
+        _.forEach(tasks, task => {
+            this.manageMessage(task);
+        });
+
+        if (this.firstCall && !_.isEmpty(tasks)) {
+            this.firstCall = false;
+            this.$rootScope.$broadcast("tasks.pending");
+        }
+
+        // refresh while there's pending task
+        if (!this.firstCall) {
+            this.$timeout(() => {
```

Comment:
interval + unsubscribe function that kill the interval if needs be

Reply:
Done :)